### PR TITLE
Send Problem report when CredEx not found

### PR DESCRIPTION
**Pull request by usingtechnology**
See #2549 

Address the situation when an issuer sends an offer and then deletes that offer (credential exchange) before the holder accepts it. When this is an `indy` credential exchange, a Problem Report is sent; when this is a `json-ld` exchange, the holder's accept actually becomes an unbound request for the credential and the flow carries on.

A discussion is warranted to see if we want to implement something more global? Seems like at any point in any protocol one side may not find their record and the other side should be notified of the abandonment. Maybe it is already done and this was just a gap...

Added BDD tests to address the deletion and followup acceptance for `indy` and `json-ld` and also the holder sending a request for credential to issuer. Again, `indy` is not allowed, `json-ld` can flow into a credential issuance.

### CODE REVIEW

File: aries_cloudagent/protocols/issue_credential/v2_0/manager.py [hunk 1]
```diff
@@ -412,6 +412,16 @@ async def receive_request(
...
         connection_id = None if oob_record else connection_record.connection_id
 
+        format_handlers = []
+        format_handlers_without_offer = []
+        for format in cred_request_message.formats:
+            f = V20CredFormat.Format.get(format.format)
+            if f:
+                format_handlers.append(f)
+                if f.handler(self.profile).can_receive_request_without_offer():
+                    format_handlers_without_offer.append(f)
```
Comment (dbluhm) [inline aries_cloudagent/protocols/issue_credential/v2_0/manager.py:422]: Good case for a list comprehension:

```python
handlers = [
    handler(self.profile)
    for format in cred_request_message.formats
    if (handler := V20CredFormat.Format.get(format.format))
]
handlers_without_offer = [
    handler for handler in handlers
    if handler.can_receive_request_without_offer()
]
```

File: aries_cloudagent/protocols/issue_credential/v2_0/manager.py [hunk 2]
```diff
@@ -420,31 +430,33 @@ async def receive_request(
...
-                    role=V20CredExRecord.ROLE_ISSUER,
-                    auto_remove=not self._profile.settings.get(
-                        "preserve_exchange_records"
-                    ),
-                    trace=(cred_request_message._trace is not None),
-                    auto_issue=self._profile.settings.get(
-                        "debug.auto_respond_credential_request"
-                    ),
-                )
+                if len(format_handlers_without_offer):
```
Comment (dbluhm) [inline aries_cloudagent/protocols/issue_credential/v2_0/manager.py:435]: Nitpick but I'd say it's more idiomatic to just do `if format_handlers_without_offer:`

File: aries_cloudagent/protocols/issue_credential/v2_0/manager.py (hunk 1 above)
Comment (usingtechnology) [reply aries_cloudagent/protocols/issue_credential/v2_0/manager.py:422]: i like it; will do.

**Comment by usingtechnology** [comment]
@dbluhm - just bumping this for another review. added your suggestions and then put a similar exception handling in the v1 code.

**Review by dbluhm — APPROVED** [review]
LGTM!

**Pull request merged**